### PR TITLE
mobile: smaller changes to login screen

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [x ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

the "hamburger" was visible in the login screen, allowing the user to bypass login, without setting the proper status.

Save password local is default false, change to true (securing the user do not need to change it, remark the login screen does not contain this checkbox).

remove a couple of unused functions.

This PR is preparation to remove a big part of qmlprefs, but due to size of PR this will come afterward.

### CODE REVIEW

**Comment by janiversen** [comment]
force pushed with latest subsurface-master, ready to be merged.

File: subsurface-mobile-main.cpp [hunk 1]
```diff
@@ -64,6 +65,7 @@ int main(int argc, char **argv)
 
 	// some hard coded settings
 	qPrefDisplay::set_animation_speed(0); // we render the profile to pixmap, no animations
+	qPrefCloudStorage::set_save_password_local(true);
```
Comment (dirkhh) [inline subsurface-mobile-main.cpp:68]: this value isn't used at all in Subsurface-mobile.
https://github.com/Subsurface-divelog/subsurface/blob/967f0e56d79c4c25e54619ef8f56770baa9edaa8/mobile-widgets/qmlmanager.cpp#L540
Therefor setting it here makes no sense, nor does the commit message that assumes that this is a thing.

File: mobile-widgets/qmlmanager.cpp [hunk 2]
```diff
@@ -537,11 +536,9 @@ void QMLManager::saveCloudCredentials()
...
 		return;
 
-	s.beginGroup("CloudStorage");
-	s.setValue("email", QMLPrefs::instance()->cloudUserName());
-	s.setValue("password", QMLPrefs::instance()->cloudPassword());
-	s.setValue("cloud_verification_status", QMLPrefs::instance()->credentialStatus());
-	s.sync();
+	qPrefCloudStorage::set_cloud_storage_email(QMLPrefs::instance()->cloudUserName());
+	qPrefCloudStorage::set_cloud_storage_password(QMLPrefs::instance()->cloudPassword());
+	qPrefCloudStorage::set_cloud_verification_status(QMLPrefs::instance()->credentialStatus());
```
Comment (dirkhh) [inline mobile-widgets/qmlmanager.cpp:541]: I see absolutely zero value in changes like this.
We have code that works. You split over two commits changes that make that code work differently.
And that I now need to spend time figuring out if it breaks something or has unintended side effects.
How is that good use of my time?

File: mobile-widgets/qml/main.qml [hunk 3]
```diff
@@ -52,8 +52,8 @@ Kirigami.ApplicationWindow {
 		id: fontMetrics
 		Component.onCompleted: {
 			manager.appendTextToLog("Using the following font: " + fontMetrics.font.family
-				    + " at " + subsurfaceTheme.basePointSize + "pt" +
-				    " with mobile_scale: " + PrefDisplay.mobile_scale)
+					+ " at " + subsurfaceTheme.basePointSize + "pt" +
+					" with mobile_scale: " + PrefDisplay.mobile_scale)
```
Comment (dirkhh) [inline mobile-widgets/qml/main.qml]: random whitespace change in the middle of an unrelated commit

**Review by dirkhh — CHANGES_REQUESTED** [review]
a couple of useful changes, mixed in with other things that I would have to think a lot more about than I really want to right now.
I really don't understand how you decide what goes into one PR. And how you split commits.
How can the first and third commit in this series not be one commit. I wouldn't want to take that commit anyway, but at least it would have made sense together.

File: subsurface-mobile-main.cpp (hunk 1 above)
Comment (janiversen) [reply subsurface-mobile-main.cpp:68]: that is actually not true. if save_password_local == false it will ask you for a password every time. it is however true it is not used in QML directly, but as mentioned indirectly.

File: mobile-widgets/qmlmanager.cpp (hunk 2 above)
Comment (janiversen) [reply mobile-widgets/qmlmanager.cpp:541]: I see where you are heading...might be better to abandon the whole thing. 

Actually the code that "works" does not respond correctly to default values, something you (thanks) introduced into qPrefs.

Comment (dirkhh) [reply mobile-widgets/qmlmanager.cpp:541]: that's a fair criticism of my comment.

I withdraw the statements that there's "absolutely zero value".

File: mobile-widgets/qml/main.qml (hunk 3 above)
Comment (janiversen) [reply mobile-widgets/qml/main.qml]: Corrected, sorry for the problem.

**Review by dirkhh — APPROVED** [review]
